relayer/src/kill_switch.rs

```diff
@@ -0,0 +1,176 @@
+use std::time::{Duration, Instant};
+
+use ethereum_client::{EthApi, MerkleRootEntry};
+use gear_rpc_client::GearApi;
+use prover::proving::GenesisConfig;
+
+use crate::{
+    proof_storage::ProofStorage,
+    prover_interface::{self, FinalProof},
+    relay_merkle_roots::{submit_proof_to_ethereum, sync_authority_set_id},
+};
+
+const MIN_MAIN_LOOP_DURATION: Duration = Duration::from_secs(12);
+
+pub struct KillSwitchRelayer {
+    gear_api: GearApi,
+    eth_api: EthApi,
+    genesis_config: GenesisConfig,
+    proof_storage: Box<dyn ProofStorage>,
+
+    // Next eth block to process.
+    from_eth_block: Option<u64>,
+    emergency_stop: bool,
```

Comment:
I think we can just `exit()` when killing tx is finalized on ethereum if there exists a simple way to setup `docker-compose` to not restart service if that happened (and restart only if it panicked)

Reply:
Fixed